### PR TITLE
Replace tableScan with values for empty table prefixes

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java [hunk 1]
```diff
@@ -164,7 +165,13 @@ public boolean usesLegacyTableLayouts()
     @Override
     public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
     {
-        return new ConnectorTableProperties();
+        InformationSchemaTableHandle tableHandle = (InformationSchemaTableHandle) table;
```
[the remark this answers is not in the document]
Comment (Praveen2112) [reply presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java]: have added a test for the same

[the remark this answers is not in the document]
Comment (kokosing) [reply presto-main/src/main/java/io/prestosql/connector/informationschema/InformationSchemaMetadata.java]: My idea for test here is to run a query and see that plan does not contain table scan. That possibly could be added to `io.prestosql.tests.TestInformationSchemaConnector`.

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java [hunk 2]
```diff
@@ -192,6 +192,12 @@ public void testMetadataCalls()
                         .withListSchemasCount(1)
                         .withListTablesCount(2)
                         .withGetColumnsCount(10000));
+        assertNoMetadataCalls("SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", "VALUES 0");
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: I was also thinking about:
```
        assertFalse(searchFrom(getQueryRunner().createPlan(getSession(), "SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", WarningCollector.NOOP).getRoot())
                .where(TableScanNode.class::isInstance)
                .findFirst()
                .isPresent());
```

Comment (Praveen2112) [reply presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: Thanks for your insights. I have applied your comments

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java [hunk 3]
```diff
@@ -192,13 +196,20 @@ public void testMetadataCalls()
...
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='tpch' AND table_name=''", "VALUES 0");
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='' AND table_name='table'", "VALUES 0");
+        assertFalse(searchFrom(getQueryRunner().createPlan(getSession(), "SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", WarningCollector.NOOP).getRoot())
+                .where(TableScanNode.class::isInstance)
+                .findFirst()
+                .isPresent());
+        assertFalse(searchFrom(getQueryRunner().createPlan(getSession(), "SELECT count(*) from test_catalog.information_schema.tables WHERE table_name = ''", WarningCollector.NOOP).getRoot())
+                .where(TableScanNode.class::isInstance)
+                .findFirst()
+                .isPresent());
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: please extract a method `assertNoTableScan("SELECT count(*) from test_catalog.information_schema.tables WHERE table_name = ''")`

Also please pass some message to `assertFalse`

File: presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java [hunk 4]
```diff
@@ -192,13 +196,14 @@ public void testMetadataCalls()
...
                         .withGetColumnsCount(10000));
+        assertNoMetadataCalls("SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''", "VALUES 0");
     }
 
     @Test
     public void testInformationForEmptyNames()
     {
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='tpch' AND table_name=''", "VALUES 0");
-        assertQuery("SELECT count(*) FROM test_catalog.information_schema.tables WHERE table_schema='' AND table_name='table'", "VALUES 0");
+        assertNoTableScan("SELECT count(*) from test_catalog.information_schema.tables WHERE table_schema = ''");
```
Comment (kokosing) [inline presto-tests/src/test/java/io/prestosql/tests/TestInformationSchemaConnector.java]: you lost `table_schema='tpch' `

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**